iceoryx_utils/source/posix_wrapper/access_control.cpp

```diff
@@ -104,43 +104,45 @@ bool AccessController::addPermissionEntry(const Category f_category,
     {
     case Category::SPECIFIC_USER:
     {
+        bool result{false};
         if (f_name.empty())
         {
             std::cerr << "Error: specific users must have an explicit name." << std::endl;
-            return false;
+            result = false;
```

Comment:
I wouldn't follow this approach and would perform here an early return which is much safer.
Could you please change it back to an early return! In the end you changed the behavior of the code and for instance, when the `f_name` is empty you are calling now `getUserID` in line 114 with an empty name which is most likely not allowed and produce even more errors.

If you discover something where you are unable to continue to work in a function you should do a return immediately otherwise you have to track the state in a separate variable and check with every additional action if it even makes sense to continue.

Early returns are a strong tool to avoid errors while implementing and we should use them as often as possible!

Reply:
I fully agree with @elfenpiff. Early returns are a good and safe style and already used extensively throughout the code base. Is MISRA the reason why this change was suggested? MISRA 2008 does not allow more than one return in a function. However, it has been agreed by many experts that this rule does not necessarily make your code safe. I'd suggest to adapt our documentation and focus on the AUTOSAR rules till the end of the year. This is what will be checked with Axivion (see #409)